lib/graphql_schema.rb

```diff
@@ -24,6 +24,10 @@ def types
     @types ||= @hash.fetch('types').map{ |type_hash| TypeDefinition.new(type_hash) }.sort_by(&:name)
   end
 
+  def type_from_name(type_name)
+    @types.find { |type| type.name == type_name }
```

Comment:
I was thinking build the entire look up table when it's `type_from_name` is used. Because again this would only improve if you continued to access the same type over and over again.

Might improve for something like `Shop` where you might access it multiple times and it's just under root. But for far off leaf nodes it's unlikely this would bring an improvement.